### PR TITLE
DrawCircleTool: Don't create empty objects

### PR DESCRIPTION
**Pull request by dg0yt**
For creating a circle, we need more than just the position of the
initial mouse press. That's why this tool leaves the preview object
empty after startDrawing(). However, the empty preview object must
not be added to the map on finishDrawing.
Fixes GH-1497 (crash after drawing with circle tool).

### CODE REVIEW

File: src/tools/draw_circle_tool.cpp [hunk 1]
```diff
@@ -223,7 +223,10 @@ void DrawCircleTool::finishDrawing()
 	second_point_set = false;
 	updateStatusText();
 	
-	DrawLineAndAreaTool::finishDrawing();
+	if (preview_path->parts().empty())
```
Comment (dg0yt) [inline src/tools/draw_circle_tool.cpp]: `|| preview_path.parts().back().empty()`

**Comment by lpechacek** [comment]
> On an afterthought, ...

The thing is that it is not safe to call the following sequence: `startDrawing(); finishDrawing();`  I'd expect a sensible API to allow this.

**Comment by lpechacek** [comment]
https://github.com/lpechacek/mapper/commits/issue-1497-draw-circle-crash-fix contains my proposal of the fix. It embodies all the above ideas, as far as I can tell.

**Comment by dg0yt** [comment]
> The thing is that it is not safe to call the following sequence: startDrawing(); finishDrawing();  I'd expect a sensible API to allow this.

We agree on this point.

> https://github.com/lpechacek/mapper/commits/issue-1497-draw-circle-crash-fix contains my proposal of the fix. It embodies all the above ideas, as far as I can tell.

That's one change to fix DrawCircleTool::finishDrawing (the special case) and another one to change DrawLineAndAreaTool::finishDrawing (the general case), right? This reflects my afterthought after reviewing my own change, and the chosen conditions look good to me.

**Comment by dg0yt** [comment]
Picked your DrawCircleTool change (but edited commit message).
Rewrote the DrawLineAndAreaTool change.

**Pull request merged**